src/commands/source/download.rs

```diff
@@ -277,39 +297,69 @@ pub async fn download(
                     }
 
                     if source_path_exists && !force {
-                        Err(anyhow!("Source exists: {}", source.path().display()))
+                        info!("Source already exists: {}", source.path().display());
+                        Ok(DownloadResult::Skipped)
                     } else {
+                        let mut download_result: Option<DownloadResult> = None;
+
                         if source_path_exists
                         /* && force is implied by 'if' above*/
                         {
                             source.remove_file().await?;
+                            download_result = Some(DownloadResult::Forced)
                         }
 
                         progressbar.lock().await.inc_download_count().await;
                         {
                             let permit = download_sema.acquire_owned().await?;
                             perform_download(&source, progressbar.clone(), timeout).await?;
                             drop(permit);
+
+                            if download_result.is_none() {
```

Comment:
This (and other places) obviously needs to be changed a bit to use delayed initialization for `DownloadResult`.